### ISSUE TITLE
How to include NA genotypes in sbam file

### ISSUE THREAD

**Issue by songtaogui** [xqwen/dap#17]
Hi,
Does dap-g Individual-level Data "sbam" format support unknown genotype for certain variant in certain sample? Or one MUST impute their genotype matrix before running dap-g?
Best wishes,
Thank you

**Comment by xqwen** [comment]
Imputation before analysis is HIGHLY recommended. By default, DAP imputes NA with the mean genotype computed from all other non-missing genotypes at the target variant.

**Comment by songtaogui** [comment]
@xqwen 
Thank you for your kind answer. 

So the sbam format for DAP accepts "NA" string to represent no data genotype, right? Have you estimated the performances between imputed/un-imputed data ( Sence I have a huge genotype data set, the imputing would be time consuming ) ?

PS. I found the `get_credible_set.pl` script unsuitable for DAP-G output. So I have modified the [L54-56 of the script](https://github.com/xqwen/dap/blob/e57f25b2310a5d7ba05415fb836059c6ddb05da5/utility/get_credible_set.pl#L54) by changing `$data[-1]` to `$data[4]` and it works fine. Am I doing the right thing?

PPS. Let's say that I have several GWAS Peaks, considering the computational complexity, it is better to run DAP-G for each Peak region, right? So is it resonable to compare the PIPs of the credible sets of all GWAS peaks to get a final causal SNP set?

PPPS. I got a non-zero exit status (In shell `echo $?` got `1`) after running dap-g with example data, but didn't get any error messages and the output file looks normal. So does the non-zero status indicated error in dap-g anyway? 

Thank you again for your help.

Best wishes,

Songtao Gui

**Comment by xqwen** [comment]
On Tue, Apr 28, 2020 at 12:10 AM Songtao Gui <notifications@github.com>
wrote:

> @xqwen <https://github.com/xqwen>
> Thank you for your kind answer.
>
> So the sbam format for DAP accepts "NA" string to represent no data
> genotype, right? Have you estimated the performances between
> imputed/un-imputed data ( Sence I have a huge genotype data set, the
> imputing would be time consuming ) ?
>

Yes. If you don't a large percentage of missing data, the performance will
be similar. However, when > 10% of the genotype data are missing from a
marker, then formal imputation will be significantly better -- I believe.




> PS. I found the get_credible_set.pl script unsuitable for DAP-G output.
> So I have modified the L54-56 of the script
> <https://github.com/xqwen/dap/blob/e57f25b2310a5d7ba05415fb836059c6ddb05da5/utility/get_credible_set.pl#L54>
> by changing $data[-1] to $data[4] and it works fine. Am I doing the right
> thing?
>


Thanks! We forgot to update the script after changing the output format.

> PPS. Let's say that I have several GWAS Peaks, considering the
> computational complexity, it is better to run DAP-G for each Peak region,
> right? So is it resonable to compare the PIPs of the credible sets of all
> GWAS peaks to get a final causal SNP set?
>

I think you should run for each peak region. I don't quite understand the
last question though.

Best,

William

**Comment by songtaogui** [comment]
Thank you,

> I think you should run for each peak region. 

Yes, I have ran for each peak region, now I was wondering that is it resonable (or is it necessary) to combine the credible SNPs of all peak regions, and run dap-g again in whole genome wide scale to 
narrow the credible set size?


> PPPS. I got a non-zero exit status (In shell echo $? got 1) after running dap-g with example data, but didn't get any error messages and the output file looks normal. So does the non-zero status indicated error in dap-g anyway?

I was trying to imbed dap-g into a shell-script-based pipeline, so I need to check if the progress was successfully finished. In linux, when a progress successfully finished, it usually returns an exit status of "0" (stored in `$?` variable). So the `non-zero exit status` usually indicated ABORTED. But I found that dap-g returns a non-zero exit status even when the log and output looks fine. So I was wondering that if this is normal or some unnoticed error occured.

Thank you,
Best wishes,

Songtao Gui
